### ISSUE TITLE
Question about the Implementation of AutoCon Loss

### ISSUE THREAD

**Issue by wuzhaow** [junwoopark92/Self-Supervised-Contrastive-Forecsating#4]
Hi, I've been studying your paper and code implementation. I noticed some differences between the AutoCon loss function described in the paper (Equation 3) and its actual implementation.

In the paper, the negative samples are selected based on the condition r(i,k)≤r(i,j), which means pairs with lower autocorrelation than the anchor pair should be treated as negative samples. However, in the code implementation (particularly in avg_global_contrastive_loss), it seems that all non-self pairs are treated as negative samples using a mask operation:

```python
neg_mask = torch.scatter(
    torch.ones_like(global_distmap),
    1,
    torch.arange(B).view(-1, 1).to(device),
    0
)

Could you clarify:

Why was this implementation approach chosen over the relative selection strategy described in the paper?

Does this simplified implementation affect the model's ability to capture long-term variations?

Are there any empirical comparisons between the two approaches?

Thank you for sharing your research. Looking forward to your response!

**Comment by JimmyMcgill0828** [comment]
Same question.

**Comment by fishfly0809** [comment]
same question

**Comment by junwoopark92** [comment]
Thank you for your interest in the paper.

As you mentioned, the code implementation is indeed a simplified version compared to Equation 3 in the paper.

In the paper's version, all pairs function as anchors once, and among all pairs except for itself, those with lower AC values than the anchor are considered negatives.

In the implementation, the pair with the highest autocorrelation is selected as the anchor, and all other pairs are treated as negatives. As a result, the loss is calculated over fewer pairs compared to the paper's version.

However, the simplified version still preserves the core intention of AutoCon loss, which is to bring the representations of pairs with higher correlation closer together than other pairs.

While the number of pairs considered within a batch is smaller in calculating AutoCon loss, we generally run multiple epochs, so it is expected to approximate the paper's version over epochs.

1. Why was this implementation approach chosen over the relative selection strategy described in the paper?
->  computationally efficient without redundant calculations.

2. Does this simplified implementation affect the model's ability to capture long-term variations?
-> The qualitative and quantitative findings were obtained using this code implementation, indicating that this implementation effectively captures long-term correlation as well.

3. Are there any empirical comparisons between the two approaches?
-> We don’t remember exactly, but we tried both implementations and recall that there wasn’t much difference.

I’m not sure if I understood the question correctly since it’s been a while since I last looked at the paper, so please let me know if there are any remaining questions.
